backend/src/dfini_hack_team4.did

```diff
@@ -0,0 +1,59 @@
+type Amount = nat64;
+type AnswerId = nat64;
+type Solution = blob;
+type TaskId = nat64;
+type TaskPayload = blob;
+type Timestamp = nat64;
+
+type TaskType = variant {
+  translate_text;
+  edit_image;
+};
+
+type TaskStatus = variant {
+  open;
+  closed;
+};
+
+type Choice = variant {
+  yes;
+  no;
+};
+
+type Vote = record {
+  voter: principal;
+  choice: Choice;
+};
+
+type Answer = record {
+  submitter: principal;
+  submission_time: Timestamp;
+  solution: Solution;
+  votes: vec Vote;
+};
+
+type Task = record {
+  submitter: principal;
+  type: TaskType;
+  payload: TaskPayload;
+  deadline: Deadline,
+  reward: Reward,
+  answers: vec Answer;
+  status: TaskStatus,
+};
+
+type ShortTask = record {
+  id: TaskId;
+  submitter: principal;
+  type: TaskType;
+};
+
+service : () -> {
+  register: () -> ();
+  submit_task : (TaskType, TaskPayload, Timestamp, Amount) -> (TaskId);
```

Comment:
Should 
```suggestion
  submit_task : (TaskType, Task, TaskPayload, Duration, Amount) -> (TaskId);
```

Reply:
* As explained above, I think we need more than a taskType,i.e., also a more detailed description of the task. 
* Maybe it would be easier to give the voting deadline as a duration rather than a timestamp and we just set the deadline to now + duration ?
* I assume Amount = rewards amount, right?

Reply:
> As explained above, I think we need more than a taskType,i.e., also a more detailed description of the task.

My thinking was that the `TaskPayload` would be a Candid encoded value of the required input. And in the backend we would use the type to attempt to decode the payload and if we couldn't we would fail. I see that I missed adding a couple things that would make this more clear, so let me add them.

> Maybe it would be easier to give the voting deadline as a duration rather than a timestamp and we just set the deadline to now + duration ?

Yes, I can change to duration for the input.

> I assume Amount = rewards amount, right?

Amount is the type that will represent a number of tokens in general. In this context, it is indeed the reward. It will be more clear in the Rust side where we'll give a proper name to the input parameter (here we only need to define its type).